src/client/opamAuxCommands.ml

```diff
@@ -304,11 +309,13 @@ let autopin_aux st ?quiet ?(for_view=false) ?recurse ?subpath atom_or_local_list
           &&
           (* For `opam show`, we need to check does the opam file changed to
              perform a simulated pin if so *)
-          (not for_view ||
+          ((not for_view && not locked) ||
```

Comment:
I wonder why `not_locked` needs to be checked here, it should already have affected the contents of either opam file (and I would expect the situation to be symmetric w.r.t. the flag ?)